### PR TITLE
Documentation for metapackages dependencies

### PR DESCRIPTION
**Pull request by asilveir**
This document aims to inform how to proceed when is necessary to add new dependencies to the repositories.

### CODE REVIEW

**Comment by vojtechsokol** [comment]
When opening PR with changes only to documentation, please add label `documentation`. This way tests and copr build will be skipped, as they are not needed.

**Comment by asilveir** [comment]
> When opening PR with changes only to documentation, please add label `documentation`. This way tests and copr build will be skipped, as they are not needed.

Thank you @vojtechsokol for the tip.
Next time I will do that.

File: docs/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+
+# How to deal with dependencies
+
+For work with dependencies in Leapp repositories, you need to understand the
+dependencies types and why Leapp has a special way to deal with dependencies.
```
Comment (pirat89) [inline docs/dependencies.md]: I would reword this as `dependencies types` is something what people will think: I understand RPM (or in general: package) dependencies. Something like:
> problem we have with dependencies itself. Read this....

File: docs/dependencies.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Read this
+[Leapp documentation](https://github.com/oamg/leapp/blob/master/docs/source/dependencies.md)
+to understand it.
+
+## What to do when dependencies needs to be changed
+
+You can add dependencies for all repositories or only for one repository
```
Comment (pirat89) [inline docs/dependencies.md]: Nope. There is nothing like that. Both things you described are now basically for all repositories. When we will add additional repository, we will have to rethink completely how to deal with packaging. Probably packaging will have to be hidden under repositories itself. The thing is whether you are resolving dependencies for target system or src system (system where you install  the leapp).

**Comment by asilveir** [comment]
> Update the _"Do not introduce new dependencies"_ section in https://github.com/oamg/leapp-repository/blob/master/docs/best-practices.md.

@bocekm I added a "for more details" in the footer I think is enough if you don't, please let me know.

**Comment by bocekm** [comment]
> @bocekm I added a "for more details" in the footer I think is enough if you don't, please let me know.

Yeah, it is. Thanks.
Is this still WIP or ready for review?

**Comment by bocekm** [comment]
PR created: https://github.com/asilveir/leapp-repository/pull/3.

**Comment by asilveir** [comment]
> It looks like you replaced the `dependencies.md` file by the one for best practices.



> It looks like you replaced the `dependencies.md` file by the one for best practices.

Ops, fixed.
